### PR TITLE
AP_HAL_ChibiOS: Add support for STM32F413xx

### PR DESCRIPTION
**Pull request by joshanne**
Generated the STM32F413xx python module from datasheet (`stm32f413cg`) and reference manual (`rm0430-stm32f413423-advanced-armbased-32bit-mcus-stmicroelectronics`).

Added changes for supporting STM32F413xx to mcuconf.

### CODE REVIEW

**Comment by joshanne** [comment]
We are using a board with an STM32F413 - Best I can see, there have been requests to add support for STM32F413xx in the past, but the pindef has never been added.

I've used the Ref Man and datasheet to pull out the CSV required to run the parser scripts.

We have a build running on the F413, seemingly without issues.

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/addfunc_parse.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def is_pin(str):
         return False
 
 def parse_adc_table(fname, table):
-    csvt = csv.reader(open(fname,'rb'))
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/addfunc_parse.py:31]: I think you should remove this drive-by

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/af_parse.py [hunk 2]
```diff
@@ -86,10 +86,10 @@ def parse_af_table(fname, table):
 
 parse_af_table(sys.argv[1], table)
 
-sys.stdout.write("AltFunction_map = {\n");
+sys.stdout.write("AltFunction_map = {\n")
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/af_parse.py:89]: ditto

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/af_parse.py [hunk 3]
```diff
@@ -86,10 +86,10 @@ def parse_af_table(fname, table):
...
 sys.stdout.write('\t# format is PIN:FUNCTION : AFNUM\n')
 sys.stdout.write('\t# extracted from %s\n' % os.path.basename(sys.argv[1]))
 for k in sorted(table.keys(), key=cmp_to_key(pin_compare)):
     s = '"' + k + '"'
     sys.stdout.write('\t%-20s\t:\t%s,\n' % (s, table[k]))
-sys.stdout.write("}\n");
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/af_parse.py:95]: ditto

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/dma_parse.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 def parse_dma_table(fname, table):
     dma_num = 1
-    csvt = csv.reader(open(fname,'rb'))
+    csvt = csv.reader(open(fname,'r'))
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/dma_parse.py:14]: ditto

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/dma_parse.py [hunk 5]
```diff
@@ -78,7 +78,7 @@ def check_full_table(table):
 sys.stdout.write('\t# format is (DMA_TABLE, StreamNum, Channel)\n')
 sys.stdout.write('\t# extracted from %s\n' % os.path.basename(sys.argv[1]))
 
-for k in sorted(table.iterkeys()):
+for k in sorted(table.keys()):
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/dma_parse.py:81]: ditto

**Review by andyp1per — CHANGES_REQUESTED** [review]
If you remove the drive-by's this looks correct to me and safe to go in

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/addfunc_parse.py (hunk 1 above)
Comment (joshanne) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/addfunc_parse.py:31]: I don't consider this to be drive by given the script doesn't run without the change, and you can't generate the table without the script.

**Review by andyp1per — APPROVED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32F413xx.py [hunk 6]
```diff
@@ -0,0 +1,858 @@
...
+	"I2C3_RX" 	:	[(1,1,1),(1,2,3)],
+	"I2C3_TX" 	:	[(1,4,3),(1,5,6)],
+	"I2CFMP1_RX"	:	[(1,3,1),(1,0,7)],
+	"I2CFMP1_TX"	:	[(1,1,2),(1,7,4)],
+	"I2S2_EXT_RX"	:	[(1,3,3)],
+	"I2S2_EXT_TX"	:	[(1,4,2)],
+	"I2S3_EXT_RX"	:	[(1,2,2),(1,0,3)],
+	"I2S3_EXT_TX"	:	[(1,5,2)],
+	"QUADSPI" 	:	[(2,7,3)],
+	"SAI1B"   	:	[(2,4,1)],
```
Comment (joshanne) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32F413xx.py:60]: Reference manual calls this out separately to `SAI1_B` - I suspect this is a typo, I'll fix this up when I have a moment.

Comment (joshanne) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32F413xx.py]: I have adjusted the csv and regenerated the table.

**Review by peterbarker — APPROVED** [review]
Not going to break anything unrelated.
